### ISSUE TITLE
SVG graphs report "No data" for dynamically added CPUs

### ISSUE THREAD

**Issue by stevekay** [sysstat/sysstat#107]
On Fedora 23 (4.4.5-300.fc23.x86_64) under VMware player 6.0.6, using the latest sysstat version (https://github.com/sysstat/sysstat/commit/d95fdd9f1faaeb31252d92cabecdef1183d3661c), SVG graphs for CPU can sometimes report "No data" after dynamically adding CPUs.  

Steps to reproduce 
- create VM with single cpu
- add 2 cpus without performing reboot
- run `./sadc -S XALL 1 60 >/var/tmp/sar-20160318.dat` to capture stats
- run `./sar -u -P ALL -f /var/tmp/sar-20160318.dat`, reports CPU usage across all cores ok

        [steve@fedora sysstat]$ ./sar -u -P ALL -f /var/tmp/sar-20160318.dat | head
        Linux 4.4.5-300.fc23.x86_64 (fedora)    03/18/2016      _x86_64_        (3 CPU)
        01:07:35 PM     CPU     %user     %nice   %system   %iowait    %steal     %idle
        01:07:36 PM     all      1.00      0.00      0.00      0.00      0.00     99.00
        01:07:36 PM       0      1.00      0.00      0.00      0.00      0.00     99.00
        01:07:36 PM       1      0.00      0.00      0.00      0.00      0.00      0.00
        01:07:36 PM       2      0.00      0.00      0.00      0.00      0.00      0.00
        01:07:37 PM     all      0.00      0.00      1.01      0.00      0.00     98.99
        01:07:37 PM       0      0.00      0.00      1.01      0.00      0.00     98.99
        01:07:37 PM       1      0.00      0.00      0.00      0.00      0.00      0.00
        [steve@fedora sysstat]$

- run `./sadf -g /var/tmp/sar-20160318.dat -- -A >foo.svg`, reports '**No data**' for the 2 "new" cpus
![sar svg lacking data on 2 cpus](http://i.imgur.com/WRL3n4g.png)
- reboot VM, rerun the `sadf` against the data file , SVG still reports 'No data'
- rerun the `sadc` to capture new data file, and then rerun the `sadf`, SVG reports all 3 cpus correctly

# Files
- http://stevekay.github.io/sar-20160318.dat 
- http://stevekay.github.io/foo.svg

**Comment by sysstat** [comment]
Hi Steve,
Thank you for this bug report (and your corresponding sar datafile which helped me to reproduce and fix the bug :-) ) Commit #2e8071e fixes it.
Note that I thought it wasn't a bug at first because your 2 added CPUs appear as offline with all the metrics set to 0.00:
```
$ ./sar -f ../tmp/sar-20160318.dat -u -P 1,2 | head
Linux 4.4.5-300.fc23.x86_64 (fedora) 	03/18/16 	_x86_64_	(3 CPU)

21:07:35        CPU     %user     %nice   %system   %iowait    %steal     %idle
21:07:36          1      0.00      0.00      0.00      0.00      0.00      0.00
21:07:36          2      0.00      0.00      0.00      0.00      0.00      0.00
21:07:37          1      0.00      0.00      0.00      0.00      0.00      0.00
21:07:37          2      0.00      0.00      0.00      0.00      0.00      0.00
21:07:38          1      0.00      0.00      0.00      0.00      0.00      0.00
21:07:38          2      0.00      0.00      0.00      0.00      0.00      0.00
21:07:39          1      0.00      0.00      0.00      0.00      0.00      0.00
```
Do you confirm they actually are offline? If so, they should not appear in the /proc/stat file.

**Comment by stevekay** [comment]
First of all, patch looks all good, repeated the steps and SVG now reflects the same output as `sar -u -P ALL`.

As you point out, the CPUs are added on-the-fly and appear in a present-but-offline state, hence 0% usage across cpu#1 and cpu#2.

After seeing SVG looking better here I've onlined the 2 cpus `echo 1 >online` and generated some workload....subsequent SVG graphs do look all good, with some workload on all 3 cpus.

Then tried a whole pile of cpu offlining/onlining while `sadc` was running, all still worked fine.  Couldn't break it!

```
steve@fedora sysstat]$ cat /proc/cpuinfo
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 42
model name      : Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz
stepping        : 7
microcode       : 0x29
cpu MHz         : 3399.616
cache size      : 6144 KB
physical id     : 0
siblings        : 1
core id         : 0
cpu cores       : 1
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 13
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts mmx fxsr sse sse2 ss syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts nopl xtopology tsc_reliable nonstop_tsc aperfmperf eagerfpu pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 x2apic popcnt aes xsave avx hypervisor lahf_lm ida arat epb pln pts dtherm xsaveopt
bugs            :
bogomips        : 6799.23
clflush size    : 64
cache_alignment : 64
address sizes   : 40 bits physical, 48 bits virtual
power management:

[steve@fedora sysstat]$ cat /proc/stat
cpu  2202 0 5412 48441 44 0 35 0 0 0
cpu0 2202 0 5412 48441 44 0 35 0 0 0
intr 120599 182 9 0 0 0 0 0 0 1 2 0 0 15 0 0 624 0 6795 284 16291 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
ctxt 333385
btime 1458401417
processes 2525
procs_running 4
procs_blocked 0
softirq 139421 2 94898 117 16341 6447 0 1653 0 0 19963
[steve@fedora sysstat]$ lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                3
On-line CPU(s) list:   0
Off-line CPU(s) list:  1,2
Thread(s) per core:    1
Core(s) per socket:    1
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 42
Model name:            Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz
Stepping:              7
CPU MHz:               3399.616
BogoMIPS:              6799.23
Hypervisor vendor:     VMware
Virtualization type:   full
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              6144K
NUMA node0 CPU(s):     0
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts mmx fxsr sse sse2 ss syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts nopl xtopology tsc_reliable nonstop_tsc aperfmperf eagerfpu pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 x2apic popcnt aes xsave avx hypervisor lahf_lm ida arat epb pln pts dtherm xsaveopt
[steve@fedora sysstat]$
```
